qiskit/providers/v2/backend.py

```diff
@@ -0,0 +1,216 @@
+        """Return the backend configuration.
+
+        Returns:
+            BackendConfiguration: the configuration for the backend.
+        """
+        return self._configuration
+
+    def properties(self):
+        """Return the backend properties.
+
+        Returns:
+            BackendProperties: the configuration for the backend. If the backend
+            does not support properties, it returns ``None``.
+        """
+        return None
+
+    def provider(self):
+        """Return the backend Provider.
+
+        Returns:
+            Provider: the Provider responsible for the backend.
+        """
+        return self._provider
+
+    def status(self):
+        """Return the backend status.
+
+        Returns:
+            BackendStatus: the status of the backend.
+        """
+        return BackendStatus(backend_name=self.name(),
+                             backend_version='1',
+                             operational=True,
+                             pending_jobs=0,
+                             status_msg='')
+
+    def name(self):
+        """Return the backend name.
+
+        Returns:
+            str: the name of the backend.
+        """
+        return self._configuration.backend_name
+
+    def __str__(self):
+        return self.name()
+
+    def __repr__(self):
+        """Official string representation of a Backend.
+
+        Note that, by Qiskit convention, it is consciously *not* a fully valid
+        Python expression. Subclasses should provide 'a string of the form
+        <...some useful description...>'. [0]
+
+        [0] https://docs.python.org/3/reference/datamodel.html#object.__repr__
+        """
+        return "<{}('{}')>".format(self.__class__.__name__, self.name())
+
+    @property
+    def options(self):
+        """Return the options for the backend
+
+        The options of a backend are the dynamic parameters defining
+        how the backend is used. These are used to control the :meth:`run`
+        method.
+        """
+        return self._options
+
+    @abstractmethod
+    def run(self, run_input, **options):
+        """Run on the backend.
+
+        This method that will return a :class:`~qiskit.providers.v2.Job` object
+        that run circuits. Depending on the backend this may be either an async
+        or sync call. It is the discretion of the provider to decide whether
+        running should  block until the execution is finished or not. The Job
+        class can handle either situation.
+
+        Args:
+            run_input (QuantumCircuit or Schedule or list): An individual or a
+                list of of :class:`~qiskit.circuits.QuantumCircuit`circuits or
+                :class:`~qiskit.pulse.Schedule` objects to run on the backend.
+                For v1 providers migrating to this first version of a v2
+                provider interface a :class:`~qiskit.qobj.QasmQobj` or
+                :class:`~qiskit.qobj.PulseQobj` object. These will be
```

Comment:
this sentence seems incomplete

Reply:
do you mean a QasmQobj and PulseQobj will be supported for now? but the type in docstring doesn't say that

Reply:
Oops yeah I missed that. I'll fix it now. 

I didn't want to put `Qobj` in the docstring type because I didn't want to encourage its use for new providers. But, I still wanted to call out anyone migrating from v1 to v2 should probably still support Qobj (but deprecate it) objects for backwards compatibility reasons.

Reply:
I updated it here: https://github.com/Qiskit/qiskit-terra/pull/5086/commits/523fba97ded10606489869d4f8392e70b7febf16 hopefully it makes more sense now